sonar-html-plugin/src/main/java/org/sonar/plugins/html/core/HtmlSensor.java

```diff
@@ -133,18 +133,25 @@ private static void saveMetrics(SensorContext context, HtmlSourceCode sourceCode
       NewIssue newIssue = context.newIssue()
         .forRule(issue.ruleKey())
         .gap(issue.cost());
-      Integer line = issue.line();
-      NewIssueLocation location = newIssue.newLocation()
-        .on(inputFile)
-        .message(issue.message());
-      if (line != null) {
-        location.at(inputFile.selectLine(line));
-      }
+      NewIssueLocation location = locationForIssue(inputFile, issue, newIssue);
       newIssue.at(location);
       newIssue.save();
     }
   }
 
+  private static NewIssueLocation locationForIssue(InputFile inputFile, HtmlIssue issue, NewIssue newIssue) {
+    NewIssueLocation location = newIssue.newLocation()
+      .on(inputFile)
+      .message(issue.message());
+    Integer line = issue.line();
+    if (issue.startColumn() != null) {
+      location.at(inputFile.newRange(issue.line(), issue.startColumn(), issue.endLine(), issue.endColumn()));
```

Comment:
Do you assume that if startColumn is not null, endLine and endColumn will not be either? Or is it fine to have two last as null? 

If it's the first case I think we should change it!

Reply:
indeed, this different cases didn't make much sense, I think it's nicer with a specialized subclass for precise issue location